WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -14,21 +31,39 @@ struct WooShippingAddPackageView: View {
             }
         }
     }
+
+    enum PackageType: CaseIterable {
+        case box, envelope
+        var name: String {
+            switch self {
+            case .box:
+                return Localization.box
+            case .envelope:
+                return Localization.envelope
+            }
+        }
+    }
+
     enum Constants {
         static let defaultVerticalSpacing: CGFloat = 16.0
+        static let saveTemplateButtonID: String = "saveTemplateButtonID"
+        static let scrollToDelay: Double = 0.5
     }
 
     @Environment(\.presentationMode) var presentationMode
 
+    @StateObject private var customPackageViewModel = WooShippingAddPackageViewModel()
+    // Holds type of selected package, it can be `custom`, `carrier` or `saved`
     @State var selectedPackageType = PackageProviderType.custom
-    var addPackageButtonDisabled: Bool {
-        for (_, value) in fieldValues {
-            if value.isEmpty {
-                return true
-            }
-        }
-        return fieldValues.count != WooShippingAddPackageDimensionView.DimensionType.allCases.count
-    }
+    // Holds selected package type when custom package is selected, it can be `box` or `envelope`
+    @State var packageType: PackageType = .box
+    // Holds value for toggle that determines if we are showing button for saving the template
+    @State var showSaveTemplate: Bool = false
+    @State var packageTemplateName: String = ""
```

Comment:
What do you think about moving these to the view model, as well? It could be helpful for unit testing, and for handling the values e.g. for `packageType` and `packageTemplateName` in network calls.

Reply:
I was planning to but left it here in case I change something, but will move it! 👍